typescript/web/src/components/core/search-input/search-input.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  forwardRef,
+  Input,
+  InputElementProps,
+  InputGroup,
+  InputGroupProps,
+  InputLeftElement,
+  InputProps,
+  InputRightElement,
+  Kbd,
+  useControllableState,
+  useMergeRefs,
+} from "@chakra-ui/react";
+import { isEmpty, isNil } from "lodash/fp";
+import {
+  createContext,
+  PropsWithChildren,
+  useCallback,
+  useContext,
+  useRef,
+} from "react";
+import { SetRequired } from "type-fest";
+import { useSearchHotkeys } from "../../../hooks";
+import { Icon, IconButton } from "../icons";
+
+export type SearchInputProps = Omit<InputGroupProps, "value" | "onChange"> & {
+  value?: string;
+  onChange?: (value: string) => void;
+  inputProps?: Omit<InputProps, "value" | "onChange">;
+};
+
+export type SearchInputState = SetRequired<
+  SearchInputProps,
+  "value" | "onChange"
+>;
+
+const SearchInputContext = createContext({} as SearchInputState);
+
+const useSearchInput = () => useContext(SearchInputContext);
+
+const SearchInputProvider = ({
+  children,
+  value: propsValue,
+  onChange: propsOnChange,
+  ...props
+}: PropsWithChildren<SearchInputProps>) => {
+  const [value, onChange] = useControllableState({
+    defaultValue: "",
+    value: propsValue,
+    onChange: propsOnChange,
+  });
+  return (
+    <SearchInputContext.Provider value={{ ...props, value, onChange }}>
+      {children}
+    </SearchInputContext.Provider>
+  );
+};
+
+const SearchInputLeft = forwardRef<InputElementProps, "input">(
+  (props: InputElementProps, ref) => (
+    <InputLeftElement ref={ref} {...props} pointerEvents="none">
+      <Icon name="search" fontSize="xl" />
+    </InputLeftElement>
+  )
+);
+
+const SearchInputValue = forwardRef<InputProps, "input">(
+  (props: InputProps, ref) => {
+    const { value, onChange, inputProps } = useSearchInput();
+    return (
+      <Input
+        {...props}
+        {...inputProps}
+        ref={ref}
+        value={value}
+        onChange={useCallback(
+          (event) => onChange(event.target.value),
+          [onChange]
+        )}
+        placeholder="Search..."
+      />
+    );
+  }
+);
+
+const ClearSearchButton = () => {
+  const { onChange } = useSearchInput();
+  return (
+    <IconButton
+      icon="close"
+      label="Clear search"
+      fontSize="xl"
+      onClick={useCallback(() => onChange(""), [onChange])}
+      cursor="pointer"
+      variant="unstyled"
+      display="flex"
+      alignItems="center"
+    />
+  );
+};
+
+const SearchInputRight = forwardRef<InputElementProps, "div">((props, ref) => {
+  const { value } = useSearchInput();
+  return (
+    <InputRightElement ref={ref} {...props} justify="flex-end">
+      {isEmpty(value) ? <Kbd>/</Kbd> : <ClearSearchButton />}
```

Comment:
Shouldn't this be based on the `focus` state ?

Reply:
Well, the thing is that when you have something typed in the search, you want the clear button to be visible, even if the search input isn't focused. We could display both Kbd and ClearSearchButton at the the same time but it'd take more visual space and feels overkill (_IMO_).